client/src/main/java/edu/snu/nemo/client/JobLauncher.java

```diff
@@ -109,36 +113,91 @@ public static void main(final String[] args) throws Exception {
     // Get DeployMode Conf
     deployModeConf = Configurations.merge(getDeployModeConf(builtJobConf), clientConf);
 
-    // Launch client main
-    runUserProgramMain(builtJobConf);
+    // Start Driver and launch user program.
+    try {
+      if (jobAndDriverConf == null || deployModeConf == null || builtJobConf == null) {
```

Comment:
Why do we need this condition check?
It looks that they cannot be null.

Reply:
It looks like that it could be that the methods assigning these variables could be returning null. This code actually existed before the PR.